deepdrivemd/sim/openmm_stream/run_openmm.py

```diff
@@ -0,0 +1,200 @@
+    sort the outliers by rmsd and use simulation task id as
+    an index into this sorted array.
+
+    """
+    if cfg.next_outlier_policy == 1:
+        cfg.pickle_db = cfg.outliers_dir + "/OutlierDB.pickle"
+
+        if not os.path.exists(cfg.pickle_db):
+            return None
+
+        if cfg.lock == "set_by_deepdrivemd":
+            cfg.lock = LockFile(cfg.pickle_db)
+
+        cfg.lock.acquire()
+        with open(cfg.pickle_db, "rb") as f:
+            db = pickle.load(f)
+        md5 = db.sorted_index[cfg.task_idx]
+        rmsd = db.dictionary[md5]
+        positions_pdb = cfg.outliers_dir + f"/{md5}.pdb"
+        velocities_npy = cfg.outliers_dir + f"/{md5}.npy"
+        shutil.copy(positions_pdb, cfg.current_dir)
+        shutil.copy(velocities_npy, cfg.current_dir)
+        shutil.copy(cfg.pickle_db, cfg.current_dir)
+        cfg.lock.release()
+
+        with open(cfg.current_dir + "/rmsd.txt", "w") as f:
+            f.write(f"{rmsd}\n")
+
+        positions_pdb = cfg.current_dir + f"/{md5}.pdb"
+        velocities_npy = cfg.current_dir + f"/{md5}.npy"
+
+        return positions_pdb, velocities_npy, rmsd, md5
+
+    elif cfg.next_outlier_policy == 0:
+        "To implement"
+        return None
+
+
+def prepare_simulation(cfg: OpenMMConfig, iteration: int, sim: omm.app.Simulation):
+    """
+    Replace positions and, with cfg.copy_velocities_p probability, velocities
+    of the current simulation state from an outlier
+    """
+    sim_dir = cfg.output_path / str(iteration)
+    sim_dir.mkdir(exist_ok=True)
+    cfg.current_dir = str(sim_dir)
+
+    outlier = next_outlier(cfg, sim)
+    if outlier is not None:
+        print("There are outliers")
+        positions_pdb, velocities_npy, rmsd, md5 = outlier
+        while True:
+            try:
+                positions = pmd.load_file(positions_pdb).positions
+                velocities = np.load(velocities_npy)
+                break
+            except Exception as e:
+                print("Exception ", e)
+                print(f"Waiting for {positions_pdb} and {velocities_npy}")
+                time.sleep(5)
+
+        sim.context.setPositions(positions)
+        if random.random() < cfg.copy_velocities_p:
+            print("Copying velocities from outliers")
+            sim.context.setVelocities(velocities)
+        else:
+            print("Generating velocities randomly")
+            sim.context.setVelocitiesToTemperature(
+                cfg.temperature_kelvin * u.kelvin, random.randint(1, 10000)
+            )
+        return True
+    else:
+        print("There are no outliers")
+        return False
+
+
+def init_input(cfg):
+    """
+    The first iteration of the simulation is initialized from pdb
+    files in cfg.initial_pdb_dir. For the given simulation the pdb file is
+    selected using simulation task_id in a round robin fashion.
+    """
+    pdb_files = glob.glob(str(cfg.initial_pdb_dir / "*.pdb"))
```

Comment:
Could update to pathlib API: `pdb_files = list(cfg.initial_pdb_dir.glob("/*.pdb"))`

Reply:
Done